msal/src/main/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplication.kt

```diff
@@ -117,6 +118,8 @@ class NativeAuthPublicClientApplication(
         //  coroutine-equivalent through this CoroutineScope.
         val pcaScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
 
+        var applicationConfig:NativeAuthPublicClientApplicationConfiguration = NativeAuthPublicClientApplicationConfiguration()
```

Comment:
This is basically a singleton, correct? We can't do this, as developers need to be able to have multiple SDK instances running in the same application (e.g. using different tenant applications and user flows).

Reply:
Note: I tried this approach a few months ago, but faced the above limitation. Which is why I refactored it into passing the configuration object to every state object initialisation.

Reply:
Why did you make this change in the first place? As I see you're still passing the configuration object to the state constructors.

Reply:
I have updated the description with this info.

Reply:
If multiple SDK instances are allowed then we need to change the signature of the APIs.

Reply:
This change has been removed.